smarts/sstudio/types.py

```diff
@@ -928,6 +978,11 @@ def __post_init__(self):
                 "Only boids can have keep_alive enabled (for persistent boids)"
             )
 
+        if not isinstance(self.zone, MapZone):
+            poly = self.zone.to_geometry(road_map=None)
+            if not poly.is_valid:
+                raise ValueError("The Zone Polygon is not a valid closed loop")
```

Comment:
Is there any context we have that we can used to determine which bubble is having an issue?